### PR TITLE
Add docker setups

### PR DESCRIPTION
**Pull request by lazyguru**
## Intro
This repo will hold docker-compose.yml files to run Sublinks for both demo and local dev purposes (eventually, production/live too). This PR is the first (major) step towards that goal.

### CURRENT STATUS
Please note, this repo currently holds configuration files intended for development purposes only. As we get closer to our initial MVP release we will provide sample files for use in a live site.

### Demo Site

- Run `docker-compose -f docker-compose.demo.yml up [-d]`

### Frontend Dev

- Run `docker-compose -f docker-compose.frontend.yml up [-d]`

NOTE: This needs to be run from a subdirectory of the frontend repo. A separate PR will be opened to replace the current `etc/dev` directory with a `docker` directory containing this repo as a submodule

### CODE REVIEW

**Review by kgilles — COMMENTED** [review]
Things look good to me after a first pass. Although I can't say I'm familiar with most things outside the frontend setup. Which did get going and was accessible for me after pointing the app volume to my local frontend root.

Only thing for now is that you definitely want to ignore node_modules/ for the seeder. That's about 600 files that don't need to be committed.

**Comment by lazyguru** [comment]
> Things look good to me after a first pass. Although I can't say I'm familiar with most things outside the frontend setup. Which did get going and was accessible for me after pointing the app volume to my local frontend root.

Cool. I'm targeting frontend repo first anyway so was hoping it would all be working. I'm intending on introducing a "docker" folder into each repo as a submodule.  That way the instructions would be more or less the same for each repo (eg, `cd docker && docker-compose -f docker-compose.<area>.yml up`)

> Only thing for now is that you definitely want to ignore node_modules/ for the seeder. That's about 600 files that don't need to be committed.

Good catch. I just pushed a commit removing it

**Comment by lazyguru** [comment]
Moving to ready for review so I can target integration into frontend repo. I will open new PRs for backend and federation configs

File: .github/CODEOWNERS
Comment (lazyguru) [inline .github/CODEOWNERS:1]: NOTE: I added a codeowners file and setup repo permissions such that based on the paths defined core-developers AND the respective team will be auto-added to any PR that modifies a file in that path

File: lemmy-ui/services.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  options:
+    max-size: "50m"
+    max-file: "4"
+
+services:
+  ui:
+    image: dessalines/lemmy-ui:0.19.3
+    environment:
+      #- LEMMY_UI_DEBUG=true
+      - LEMMY_UI_HOST=0.0.0.0:3000
```
Comment (lazyguru) [inline lemmy-ui/services.yml:14]: NOTE: I changed from the default 1234 to 3000. This was to match the sublinks-ui and so I could reference either container as "ui" for purposes of the nginx config files. Only one of the 2 containers can be run (but not sure why you would ever need to run both?)

File: README.md [hunk 2]
```diff
@@ -1,2 +1,22 @@
...
+
+CURRENT STATUS: Please note, this repo currently holds configuration files intended for development purposes only. As we get closer to our initial MVP release we will provide sample files for use in a live site.
+
+## Demo Site
+
+Run `docker-compose -f docker-compose.demo.yml up [-d]`
+
+## Frontend Dev
+
+Run `docker-compose -f docker-compose.frontend.yml up [-d]`
```
Comment (kgilles) [inline README.md:35]: `[-d]` aborts the command for me with `no matches found: [-d]`.
Command succeeds if I use `-d` without the brackets.

File: common/seeder/package.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+  "name": "sublinks-job-seeder",
+  "version": "1.0.0",
+  "scripts": {
+    "seed": "node ./job-seeder.js"
+  },
+  "dependencies": {
+    "sublinks-js-client": "^0.0.3"
```
Comment (kgilles) [inline common/seeder/package.json]: Version 0.0.13 is the latest one. This could be a good opportunity to upgrade the seeder dependency.

**Review by kgilles — CHANGES_REQUESTED** [review]
Couple of small things noted.

Would it be easy enough to add some kind of option/configuration to allow specifying the platform for the Sublinks API and Federation images. As I mentioned in the Fed chat I can't run this as-is. I need to set the platform versions since these two images aren't pushed with a matching linux arch.

**Comment by lazyguru** [comment]
I'm fixing that in the individual repos (to build and push multi-arch images). Already have a PR open in federation

File: README.md (hunk 2 above)
Comment (lazyguru) [reply README.md:35]: The brackets indicate optional command arguments. I'll add a comment below each command to clarify

Comment (kgilles) [reply README.md:35]: TIL

Comment (lazyguru) [reply README.md:35]: I updated the README. Please let me know if that helps?

File: common/seeder/package.json (hunk 3 above)
Comment (lazyguru) [reply common/seeder/package.json]: Updated. Also added Dependabot config to (hopefully) keep it updated

**Review by kgilles — APPROVED** [review]
Beautiful

**Pull request merged**